Subject: [PATCH] Fix defaulting for Named networks
---
Index: webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go b/webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go
--- a/webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go	(revision de75746a9505ef3161172d99b735d6593c54f0c5)
+++ b/webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go	(date 1712596700934)
@@ -443,9 +443,9 @@
 
 	// The networkInterface CR name ("vmName-networkName-interfaceName" or "vmName-interfaceName") needs to be a DNS1123 Label
 	if networkName != "" {
-		networkIfCRName = fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceSpec.Name)
+		networkIfCRName = strings.ToLower(strings.ReplaceAll(fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceSpec.Name), " ", "-"))
 	} else {
-		networkIfCRName = fmt.Sprintf("%s-%s", vmName, interfaceSpec.Name)
+		networkIfCRName = strings.ToLower(strings.ReplaceAll(fmt.Sprintf("%s-%s", vmName, interfaceSpec.Name), " ", "-"))
 	}
 
 	for _, msg := range validation.NameIsDNSSubdomain(networkIfCRName, false) {
